app/models/employee.rb

```diff
@@ -2,50 +2,81 @@
+  accepts_nested_attributes_for :tenures,
+    reject_if: proc { |attributes| attributes['start_date'].blank? }, allow_destroy: true
 
   validates :first_name, presence: true
   validates :last_name, presence: true
-  validates :start_date, presence: true
   validates :starting_salary, presence: true
 
   default_scope { order :first_name }
-  scope :past, -> { where 'end_date < :today', today: Time.zone.today }
+  scope :past, -> { joins(:tenures).where 'tenures.end_date < :today', today: Time.zone.today }
   scope :current, lambda {
-    where 'start_date <= :today AND (end_date IS NULL OR end_date >= :today)',
-          today: Time.zone.today
+    joins(:tenures).where 'tenures.start_date <= :today' \
+          ' AND (tenures.end_date IS NULL OR tenures.end_date >= :today)', today: Time.zone.today
   }
-  scope :future, -> { where 'start_date > :today', today: Time.zone.today }
+  scope :future, -> { joins(:tenures).where 'tenures.start_date > :today', today: Time.zone.today }
   scope :non_current, lambda {
-    where 'start_date > :today OR end_date < :today', today: Time.zone.today
+    joins(:tenures)
+      .where('tenures.start_date = (SELECT MAX(tenures.start_date) ' \
+             'FROM tenures WHERE tenures.employee_id = employees.id)')
+      .group('employees.id')
+      .where('tenures.start_date > :today OR tenures.end_date < :today', today: Time.zone.today)
   }
   scope :billed, -> { where billable: true }
   scope :support, -> { where billable: false }
 
-  def self.current
-    where 'start_date <= :today AND (end_date IS NULL OR end_date >= :today)',
-          today: Time.zone.today
-  end
-
-  def self.past_or_current
-    where '(start_date <= :today AND (end_date IS NULL OR end_date >= :today))' \
-          ' or end_date < :today', today: Time.zone.today
-  end
+  scope :past, lambda {
+    joins(:tenures)
+      .where('tenures.start_date = (SELECT MAX(tenures.start_date) ' \
+             'FROM tenures WHERE tenures.employee_id = employees.id)')
+      .group('employees.id')
+      .where 'tenures.end_date < :today', today: Time.zone.today }
+  scope :current, lambda {
+    joins(:tenures).where 'tenures.start_date <= :today' \
+          ' AND (tenures.end_date IS NULL OR tenures.end_date >= :today)', today: Time.zone.today
+  }
+  scope :past_or_current, lambda {
+    joins(:tenures)
+      .where('tenures.start_date = (SELECT MAX(tenures.start_date) ' \
+             'FROM tenures WHERE tenures.employee_id = employees.id)')
+      .group('employees.id')
+      .where('(tenures.start_date <= :today' \
+          ' AND (tenures.end_date IS NULL OR tenures.end_date >= :today))' \
+          ' OR tenures.end_date < :today', today: Time.zone.today)
+  }
 
   def self.current_or_future
-    where '(start_date <= :today AND (end_date IS NULL OR end_date >= :today))' \
-          ' or start_date > :today', today: Time.zone.today
+    joins(:tenures).where '(tenures.start_date <= :today' \
```

Comment:
This and the one below seem like they would be more consistent with the other code as scopes?